app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java

```diff
@@ -835,16 +881,19 @@ private void showRoundedThumbnail(ItemViewHolder viewHolder, IconDataParcelable
 
         View iconBackground = getBoolean(PREFERENCE_USE_CIRCULAR_IMAGES)? viewHolder.genericIcon:viewHolder.iconLayout;
 
-        iconBackground.setBackgroundColor(iconSkinColor);
+        //iconBackground.setBackgroundColor(iconSkinColor);
         viewHolder.genericIcon.setVisibility(View.VISIBLE);
         GlideApp.with(mainFrag).load(iconData.loadingImage).into(viewHolder.genericIcon);
         view.setVisibility(View.INVISIBLE);
 
-        modelProvider.getPreloadRequestBuilder(iconData).listener(new RequestListener<Drawable>() {
+        RequestListener<Drawable> requestListener = new RequestListener<Drawable>() {
             @Override
             public boolean onLoadFailed(@Nullable GlideException e, Object model, Target target, boolean isFirstResource) {
                 iconBackground.setBackgroundColor(grey_color);
-                GlideApp.with(mainFrag).load(R.drawable.ic_broken_image_white_24dp).into(viewHolder.genericIcon);
+                new Handler(msg -> {
+                    GlideApp.with(mainFrag).load(R.drawable.ic_broken_image_white_24dp).into(viewHolder.genericIcon);
+                    return false;
+                }).obtainMessage().sendToTarget();
```

Comment:
Where are you setting the target? Have you tested this code? Why use a Handler?

Reply:
Yes I've tested. Handlers are necessary otherwise there was an error to not post another load in listeners.